inttest/capi-docker/capi_docker_test.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package capidocker
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/k0sproject/k0smotron/inttest/util"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/k0sproject/k0s/inttest/common"
+	k0stestutil "github.com/k0sproject/k0s/inttest/common"
+)
+
+type CAPIDockerSuite struct {
+	suite.Suite
+	client           *kubernetes.Clientset
+	restConfig       *rest.Config
+	clusterYamlsPath string
+}
+
+func TestCAPIDockerSuite(t *testing.T) {
+	s := CAPIDockerSuite{}
+	suite.Run(t, &s)
+}
+
+func (s *CAPIDockerSuite) SetupSuite() {
+	kubeConfigPath := os.Getenv("KUBECONFIG")
+	s.Require().NotEmpty(kubeConfigPath)
+	// Get kube client from kubeconfig
+	restCfg, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	// t.Logf("restCfg.Host: %s", restCfg.Host)
+	// t.Logf("restCfg full:\n%+v", restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(restCfg)
+	s.restConfig = restCfg
+
+	// Get kube client from kubeconfig
+	kubeClient, err := kubernetes.NewForConfig(restCfg)
+	s.Require().NoError(err)
+	s.Require().NotNil(kubeClient)
+	s.client = kubeClient
+
+	tmpDir := s.T().TempDir()
+	s.clusterYamlsPath = tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(s.clusterYamlsPath, []byte(dockerClusterYaml), 0644))
+}
+
+func (s *CAPIDockerSuite) TestCAPIDocker() {
+
+	// Dump the cluster yaml to a file
+	tmpDir := s.T().TempDir()
+	tmpFile := tmpDir + "/cluster.yaml"
+	s.Require().NoError(os.WriteFile(tmpFile, []byte(dockerClusterYaml), 0644))
+
+	// Apply the child cluster objects
+	s.applyClusterObjects()
+	defer func() {
+		keep := os.Getenv("KEEP_AFTER_TEST")
+		if keep == "true" {
+			return
+		}
+		if keep == "on-failure" && s.T().Failed() {
+			return
+		}
+		s.T().Log("Deleting cluster objects")
+		s.deleteCluster()
+	}()
+	s.T().Log("cluster objects applied, waiting for cluster to be ready")
+
+	// Wait for the cluster to be ready
+	// Wait to see the CP pods ready
+	s.Require().NoError(common.WaitForStatefulSet(context.TODO(), s.client, "kmc-docker-test", "default"))
```

Comment:
There are a few context.TODO().

Perhaps we could make k0s newSuiteContext public and use that or since it's a single function without further dependencies just copy and paste it.

Reply:
Can we just use `s.Context()`?

Reply:
> Can we just use s.Context()?

There is no such thing as this is NOT using the "normal" FootlooseSuite

Reply:
ok. I "borrowed" k0s suite ctx funcs for now. As there's no real teardown stuff currently as in k0s side, I'll leave it rather simple. When we further develop new cases/suites for CAPI parts we can maybe start to extract some of the stuff into a common base or somethings. Like we've done for `FootlooseSuite` in k0s side.

Reply:
@juanluisvaladas PTAL